numpy/_core/src/umath/loops_umath_fp.dispatch.c.src

```diff
@@ -250,6 +251,25 @@ NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(@TYPE@_@func@)
         simd_@intrin@_@sfx@(src1, ssrc1, src2, ssrc2, dst, sdst, len);
         return;
     }
+#endif
+#if @ispower@
```

Comment:
The logic of the nested if else conditions is getting a little hard to read. I propose splitting the `atan2` and `pow` function to make this more readable.